src/ansys/simai/core/data/model_configuration.py

```diff
@@ -0,0 +1,434 @@
+
+                    | *debug*: < 30 min, only 4 dat
+
+                    | *short*: < 24 hours
+
+                    | *default*: < 2 days, default value.
+
+                    | *long*: < 1 week
+        continuous: indicates if continuous learning is enabled. Default is False.
+        input: the inputs of the model.
+        output: the outputs of the model.
+        global_coefficients: the Global Coefficients of the model.
+        domain_of_analysis: The Domain of Analysis of the model configuration.
+
+    Example:
+        Define a new configuration and launch a build.
+
+        .. code-block:: python
+
+            import ansys.simai.core as asc
+            from ansys.simai.core.data.model_configuration import (
+                DomainAxisDefinition,
+                DomainOfAnalysis,
+                ModelConfiguration,
+                ModelInput,
+                ModelOutput,
+            )
+
+            simai = asc.from_config()
+
+            # Get the project of interest
+            aero_dyn_project = simai.projects.get(name="aero-dyn")
+
+            # Define the inputs of the model
+            model_input = ModelInput(surface=["Velocity"], boundary_conditions=["Vx"])
+
+            # Define the outputs of the model
+            model_output = ModelOutput(
+                surface=["Pressure", "WallShearStress_0"], volume=["Velocity_0", "Pressure"]
+            )
+
+            # Define the model coefficient
+            global_coefficients = [("max(Pressure)", "maxpress")]
+
+            # Set the Domain of Analysis
+            doa = DomainOfAnalysis(
+                length=("relative_to_max", 5, 8.1),
+                width=("relative_to_max", 5, 8.1),
+                height=("absolute", -4.5, 0.1),
+            )
+
+            # Define the build configuration for the model
+            new_conf = ModelConfiguration(
+                project=aero_dyn_project,
+                build_preset="debug",
+                continuous=False,
+                input=model_input,
+                output=model_output,
+                global_coefficients=global_coefficients,
+                domain_of_analysis=doa,
+            )
+
+            # Launch a mode build with the new configuration
+            new_model = simai.models.build(new_conf)
+    """
+
+    project: "Optional[Project]" = None
+    build_preset: Literal[
+        "debug",
+        "short",
+        "default",
+        "long",
+    ] = "default"
+    continuous: bool = False
+    input: ModelInput = field(default_factory=lambda: ModelInput())
+    output: ModelOutput = field(default_factory=lambda: ModelOutput())
+    domain_of_analysis: DomainOfAnalysis = field(default_factory=lambda: DomainOfAnalysis())
+
+    def __set_gc(self, gcs: list[GlobalCoefficientDefinition]):
+        verified_gcs = []
+
+        for gc in gcs:
+            gc_unit = GlobalCoefficientDefinition(*gc) if isinstance(gc, tuple) else gc
+            if self.project is not None:
```

Comment:
So if I don't set project and set GC after creating my config, my formulas won't be verified, correct ?
I think this is potential issue as right now the backend doesn't check the formulas, they're assumed to be correct/verified by the client (which might not be a great assumption but it's convenient for now).

Reply:
would it be preferable if 'project' was required variable?

Reply:
so I made `project` a required argument for model configuration. I also introduced an exception there, to enforce that a project should be defined.